crates/services/executor/src/executor.rs

```diff
@@ -1044,302 +1037,223 @@ where
             coinbase_contract_id,
         );
 
-        let mut vm = Interpreter::with_storage(
-            vm_db,
-            InterpreterParams::new(gas_price, &self.consensus_params),
-        );
+        fuel_vm::interpreter::contract::balance_increase(
+            &mut vm_db,
+            &mint.input_contract().contract_id,
+            mint.mint_asset_id(),
+            *mint.mint_amount(),
+        )
+        .map_err(|e| format!("{e}"))
+        .map_err(ExecutorError::CoinbaseCannotIncreaseBalance)?;
+        sub_block_db_commit.commit()?;
 
-        let gas_costs = self.consensus_params.gas_costs();
-        let fee_params = self.consensus_params.fee_params();
+        let block_height = *header.height();
+        let output = *mint.output_contract();
+        let mut outputs = [Output::Contract(output)];
+        self.persist_output_utxos(
+            block_height,
+            execution_data,
+            coinbase_id,
+            block_st_transaction,
+            inputs,
+            outputs.as_slice(),
+        )?;
+        self.compute_state_of_not_utxo_outputs(
+            outputs.as_mut_slice(),
+            inputs,
+            *coinbase_id,
+            block_st_transaction,
+        )?;
+        let Input::Contract(input) = core::mem::take(&mut inputs[0]) else {
+            unreachable!()
+        };
+        let Output::Contract(output) = outputs[0] else {
+            unreachable!()
+        };
+        Ok((input, output))
+    }
 
-        let ready_tx = checked_tx
-            .clone()
-            .into_ready(gas_price, gas_costs, fee_params)?;
+    fn check_mint_amount(mint: &Mint, expected_amount: u64) -> ExecutorResult<()> {
+        if *mint.mint_amount() != expected_amount {
+            return Err(ExecutorError::CoinbaseAmountMismatch)
+        }
+        Ok(())
+    }
 
-        let vm_result: StateTransition<_> = vm
-            .transact(ready_tx)
-            .map_err(|error| ExecutorError::VmExecution {
-                error: error.to_string(),
-                transaction_id: tx_id,
-            })?
-            .into();
-        let reverted = vm_result.should_revert();
+    fn check_gas_price(mint: &Mint, expected_gas_price: Word) -> ExecutorResult<()> {
+        if *mint.gas_price() != expected_gas_price {
+            return Err(ExecutorError::CoinbaseGasPriceMismatch)
+        }
+        Ok(())
+    }
 
-        let (state, mut tx, receipts): (_, Tx, _) = vm_result.into_inner();
-        #[cfg(debug_assertions)]
-        {
-            tx.precompute(&self.consensus_params.chain_id())?;
-            debug_assert_eq!(tx.id(&self.consensus_params.chain_id()), tx_id);
+    fn check_mint_has_expected_index(
+        checked_mint: &Checked<Mint>,
+        execution_data: &ExecutionData,
+    ) -> ExecutorResult<()> {
+        if checked_mint.transaction().tx_pointer().tx_index() != execution_data.tx_count {
+            return Err(ExecutorError::MintHasUnexpectedIndex)
         }
+        Ok(())
+    }
```

Comment:
I think adding 3 new methods, 3 lines each, is overkill since the `if` statement and return error type are already descriptive enough.

Could you return it back, please=)

Reply:
I strongly prefer the new code. I think the reduction of noise and keeping the caller function as flat as possible is worth the cost of maintaining another function. These benefits add up quickly and I'd like to go farther tbh but I think it's time to wrap up this PR.